nat-lab/tests/telio.py

```diff
@@ -1211,6 +1213,83 @@ async def probe_pmtu(self, host: str, expected: int):
         await self._write_command(["pmtu", host])
         await ev.wait()
 
+    async def get_coredump_pattern(self) -> tuple[str, str]:
+        process = await self._connection.create_process(
+            ["sysctl", "kernel.core_pattern"]
+        ).execute()
+        # sysctl returns a pattern that can contain just a file name, or with absolute path, like
+        # `kernel.core_pattern = /var/crash/core-%h-%e-%s-%p` or simply `kernel.core_pattern = core`
+        core_pattern = process.get_stdout().split("= ", 1)[1].rsplit("/", 1)
+
+        path = "."
+        # if there is only one element it means that there is no path present in pattern
+        if len(core_pattern) > 1:
+            path = core_pattern[0]
+        # remove the filename flags 'core-%h-%e-%s-%p'
+        file_name = core_pattern[-1].split("%", 1)[0]
+
+        return path, file_name
+
+    async def clear_core_dumps(self):
+        if (os.environ.get("NATLAB_SAVE_LOGS") is None) or (
+            self._connection.target_os != TargetOS.Linux
+        ):
+            return
+        path, file_name = await self.get_coredump_pattern()
+
+        # clean the temporary core dump folder
+        await self._connection.create_process(
+            ["rm", "-rf", "/var/core_dumps"]
+        ).execute()
+
+        # clear the system core dumps
+        if path != ".":
+            await self._connection.create_process(["rm", "-rf", path]).execute()
+            await self._connection.create_process(["mkdir", path]).execute()
+        else:
+            await self._connection.create_process(
+                ["rm", "-f", f"{file_name}*"]
+            ).execute()
+
+    async def collect_core_dumps(self):
+        if (os.environ.get("NATLAB_SAVE_LOGS") is None) or (
+            self._connection.target_os != TargetOS.Linux
+        ):
+            return
+        path, file_prefix = await self.get_coredump_pattern()
+
+        process = await self._connection.create_process(
+            ["find", path, "-maxdepth", "1", "-name", f"{file_prefix}*"]
+        ).execute()
+        files = process.get_stdout().strip().split()
+
+        process = self._connection.create_process(["cat", "/etc/hostname"])
+        await process.execute()
```

Comment:
Nit: another consistency thing, other lines create and execute the process on the same line but here you have it on separate lines